source/Nevermore.IntegrationTests/Advanced/UseExistingTransactionFixture.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Nevermore.Advanced;
+using Nevermore.IntegrationTests.SetUp;
+using NUnit.Framework;
+
+namespace Nevermore.IntegrationTests.Advanced;
+
+#pragma warning disable NV0008 // Nevermore transaction is never disposed - false positives
+
+public class UseExistingTransactionFixture : FixtureWithRelationalStore
+{
+    CancellationToken cancellationToken;
+
+    public override void SetUp()
+    {
+        base.SetUp();
+        cancellationToken = new CancellationTokenSource(TimeSpan.FromMinutes(1)).Token;
+    }
+
+    class TestConnectionAndTransaction : IConnectionAndTransaction
+    {
+        public TestConnectionAndTransaction(DbConnection connection, DbTransaction transaction)
+        {
+            Connection = connection;
+            Transaction = transaction;
+        }
+
+        public DbConnection Connection { get; }
+
+        public DbTransaction Transaction { get; }
+
+        public int CommitTransactionCalled { get; private set; }
```

Comment:
The reason I'm asserting on calls to these methods is specifically because the implementation of `IConnectionAndTransaction` is consumer-provided; it comes from outside Nevermore, so it could be anything. Given that, I care about how many times they're called a lot more than I care what they actually do, and so not asserting on outcomes was a deliberate choice.

I made them "work" by calling `Commit`/`Dispose` only for demonstration & convenience purposes honestly, these tests would work equally well if they just incremented those counters and didn't actually do any work, leaving the actual dispose up to the tests themselves.

Reply:
Makes sense, thanks.

Maybe consider the List<string> approach rather than counters? But not a big deal either way.

Reply:
I think the list approach is reasonable and has worthwhile benefits, I'll give it a go 🙂

Reply:
I've removed `IConnectionAndTransaction` now so it's no longer necessary, I'm asserting on the behaviour instead now :)